todo-app/src/Main.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import todoItems from "./todos.json";
+import ListItem from "./ListItem";
+import TodoInput from "./TodoInput";
+
+const todoIDs = todoItems.map(todo => todo.id);
+
+export default class Main extends React.Component {
+
+    state = {
+        todos: todoItems,
+        nextId: Math.max(...todoIDs) + 1
+    }
+
+    toggleCheck = todoId => {
+        const todoItems = this.state.todos.map(todo => (
+            todo.id === todoId
+                ? { ...todo, done: !todo.done }
+                : todo
+        ));
+        this.setState({ todos: todoItems });
+    }
+
+    addTodo = (description, deadline) => {
+        const todoItems = this.state.todos.map(todo => todo);
+        const id = this.state.nextId;
+        this.setState({
+            todos: [
+                ...todoItems,
```

Comment:
Instead of creating the variables using map, you can do the same like this  `...this.state.todos,`